server/pulp/server/webservices/views/search.py

```diff
@@ -226,6 +226,16 @@ def _trim_results(model, results, only):
     min_fields = set(['_id', 'id', '_href'])
     required_fields = set([field for field, val in model._fields.items() if val.required])
     return_fields = set(only) | min_fields | required_fields
+    # remove subfield from return fields names
+    return_fields_list = list(return_fields)
+    for i in range(len(return_fields_list)):
+        f = return_fields_list[i]
+        if "." in f:
+            return_fields_list[i] = f.split(".")[0]
+    return_fields = set(return_fields_list)
```

Comment:
`in` works for sets as well as lists, no need to return to a set here. There is the possibility that if "notes" and "notes.subfield" was passed, that notes will exist in the list twice, but that doesn't cause a problem.